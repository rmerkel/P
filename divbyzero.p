var x, y, z : integer;

begin
	y = 10;
	z = 0;

	x = y / 2;
	x = y / z	{	opps!	}
end.

