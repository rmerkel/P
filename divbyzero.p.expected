# divbyzero.p, 2: var x, y, z : integer;
    0: call 0, 2
    1: halt
# divbyzero.p, 3: 
# divbyzero.p, 4: begin
    2: enter 3
# divbyzero.p, 5: 	y = 10;
    3: pushConst 10
    4: pushVar 0, 5
    5: assign
# divbyzero.p, 6: 	z = 0;
    6: pushConst 0
    7: pushVar 0, 6
    8: assign
# divbyzero.p, 7: 
# divbyzero.p, 8: 	x = y / 2;
    9: pushVar 0, 5
   10: eval
   11: pushConst 2
   12: divi
   13: pushVar 0, 4
   14: assign
# divbyzero.p, 9: 	x = y / z	{	opps!	}
   15: pushVar 0, 5
   16: eval
# divbyzero.p, 10: end.
   17: pushVar 0, 6
   18: eval
   19: divi
   20: pushVar 0, 4
   21: assign
   22: ret
#divbyzero.p, 10: 

        9:         10
       10:          0
        8:          5
Attempt to divide by zero @ pc (19)!
./pl0c: runtime error: divideByZero!
